assessment-api/assessment-service/app/controllers/v4/QuestionSetController.scala

```diff
@@ -169,4 +170,35 @@ class QuestionSetController @Inject()(@Named(ActorNames.QUESTION_SET_ACTOR) ques
 		setRequestContext(questionSetRequest, version, objectType, schemaName)
 		getResult(ApiId.COPY_QUESTION_SET, questionSetActor, questionSetRequest)
 	}
+
+	def updateComment(identifier: String) = Action.async { implicit request =>
+		val headers = commonHeaders()
+		val body = requestBody()
+
+		val commentList = body.getOrDefault("comments", new java.util.ArrayList[java.util.Map[String, Object]] {}).asInstanceOf[java.util.ArrayList[java.util.Map[String, Object]]].asScala.toList
+
+		val filteredComments = commentList.groupBy(_.getOrElse("identifier", "")).values.map(_.last).toList
```

Comment:
why do we need this processing?

Reply:
According to our requirement, there can be multiple {identifier-comment} pairs which can have the same identifier. Meaning multiple comments being passed for same questionSet. We just need the last one in the request to be retained and everything else discarded. That's what this processing does.